src/ex_docmd.c

```diff
@@ -460,6 +460,37 @@ restore_dbg_stuff(struct dbg_stuff *dsp)
 }
 #endif
 
+// @param fnum A buffer number. 0 == current buffer, 1-or-more must be a valid buffer ID.
+// @param ffname The full path to where a buffer lives on-disk or would live on-disk.
+// @return if `ffname` is the same as `fnum` buffer, return true.
+//
+static int is_other_file(int fnum, char_u *ffname)
+{
+  if (fnum != 0)
+  {
+    if (fnum == curbuf->b_fnum)
+      return FALSE;
+
+    return TRUE;
+  }
+
+  if (ffname == NULL)
+    return TRUE;
+
+  if (*ffname == NUL)
+    return FALSE;
```

Comment:
I am slightly confused, why is `ffname == NULL` considered TRUE and `*fname == NUL` considered false?

Reply:
From experimentation,

1. You're right, if `fnum` is `0` and `ffname == NULL` then it would make sense to return `FALSE`, which works for all commands except `enew`.
2. `Test_edit_no_arguments` uses `*ffname == NUL`. When no argument is provided and `fnum` is 0 then that indicates that the user is refreshing the buffer, hence `TRUE`.

IMO the logic of (2) makes sense. (1) is a bit of a stretch though. How about this? We can `ffname == NULL` and return `TRUE` which wouldn't work with `enew` (`Test_enew`) but it will work if we add this check to `ex_edit`

```c
  if (eap->cmdidx == CMD_enew
      && curbuf->b_ffname != NULL
      && *curbuf->b_ffname != NUL
      && !check_can_set_curbuf_forceit(eap->forceit)
  )
    // If the user is calling `:enew` on a buffer that was named,
    // they're trying to switch to a different buffer.
    // Which triggers a 'winfixbuf' error unless [!] is included.
    return;
```

With the check above, an extra benefit is that

```
enew
set winfixbuf
enew
```

doesn't error anymore (currently it does). Another edge case that I just found and need to write a unittest for. Does that sound good to you?

Reply:
That should be fine, since the second `enew` re-uses the buffer. I suppose it would error out, if you make some changes, and then `:enew`?

Reply:
Oh I hadn't tried

```
enew
set winfixbuf
do changes
enew again
```

Another edge case. I can make another PR for that